plugins/inputs/nginx_plus/nginx_plus.go

```diff
@@ -0,0 +1,539 @@
+package nginx_plus
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+type NginxPlus struct {
+	Urls []string
+
+	client *http.Client
+
+	ResponseTimeout internal.Duration
+}
+
+var sampleConfig = `
+  ## An array of ngx_http_status_module or status URI to gather stats.
+  urls = ["http://localhost/status"]
+
+  # HTTP response timeout (default: 5s)
+  response_timeout = "5s"
+`
+
+func (n *NginxPlus) SampleConfig() string {
+	return sampleConfig
+}
+
+func (n *NginxPlus) Description() string {
+	return "Read Nginx Plus' full status information (ngx_http_status_module)"
+}
+
+func (n *NginxPlus) Gather(acc telegraf.Accumulator) error {
+	var wg sync.WaitGroup
+
+	for _, u := range n.Urls {
+		addr, err := url.Parse(u)
+		if err != nil {
+			acc.AddError(fmt.Errorf("Unable to parse address '%s': %s", u, err))
+		}
+
+		wg.Add(1)
+		go func(addr *url.URL) {
+			defer wg.Done()
+			acc.AddError(n.gatherUrl(addr, acc))
+		}(addr)
+	}
+
+	wg.Wait()
+	return nil
+}
+
+func (n *NginxPlus) gatherUrl(addr *url.URL, acc telegraf.Accumulator) error {
+	if n.ResponseTimeout.Duration < time.Second {
+		n.ResponseTimeout.Duration = time.Second * 5
+	}
+
+	client := &http.Client{
+		Timeout: n.ResponseTimeout.Duration,
+	}
```

Comment:
Initialize the client only once, probably above since it can be shared between goroutines.  Use a new http.Transport, I'd just follow the original nginx plugin here: https://github.com/influxdata/telegraf/blob/master/plugins/inputs/nginx/nginx.go#L63-L69.  If you don't have TLS support then the transport can be initialized like `&Transport{}`

The reason for this is that each http.Transport/http.Client comes with it's own connection pool and this will allow the connection to be kept alive between Gather intervals.